internal/logic_handlers/organizationlogic/service_test.go

```diff
@@ -0,0 +1,680 @@
+				mockTxProvider.TxReturns(nil, errors.New(mockDbReturnsErr))
+
+				return &dependencies{
+					Persistor:  &organizationlogicfakes.FakePersistor{},
+					TxProvider: &mockTxProvider,
+					Logger:     mockLogger,
+					Cleanup:    cleanup,
+				}, cleanup
+			},
+			mutations: func(t *testing.T, db *sqlx.DB) (toUpdatedOrganization *model.UpdateOrganization) {
+				toUpdateOrganization := &model.UpdateOrganization{
+					Id: rand.Int(),
+					Name: null.String{
+						String: "Demby",
+						Valid:  true,
+					},
+					UserId: null.Int{
+						Int:   rand.Int(),
+						Valid: true,
+					},
+				}
+
+				return toUpdateOrganization
+			},
+			assertions: func(t *testing.T, params *model.UpdateOrganization, organization *model.Organization, err error) {
+				assert.Error(t, err, "unexpected error")
+				assert.Nil(t, organization, "unexpected nil organization")
+			},
+		},
+		{
+			name:            "fail-internal-server-error",
+			ctx:             context.TODO(),
+			getDependencies: getConcreteDependencies,
+			mutations: func(t *testing.T, db *sqlx.DB) (toUpdatedOrganization *model.UpdateOrganization) {
+				testhelper.DropTable(t, db, mysqlmodel.TableNames.Organization)
+				return nil
+			},
+			assertions: func(t *testing.T, params *model.UpdateOrganization, organization *model.Organization, err error) {
+				assert.Error(t, err, "unexpected error")
+				assert.Nil(t, organization, "unexpected nil organization")
+			},
+		},
+	}
+}
+
+func TestService_UpdateOrganization(t *testing.T) {
+	for _, tt := range getTestCasesUpdateOrganizations() {
+		t.Run(tt.name, func(t *testing.T) {
+			_dependencies, cleanup := tt.getDependencies(t)
+			defer cleanup()
+
+			svc, err := New(&Config{
+				TxProvider: _dependencies.TxProvider,
+				Logger:     _dependencies.Logger,
+				Persistor:  _dependencies.Persistor,
+			})
+			require.NoError(t, err, "unexpected new error")
+
+			updateOrganizationStruct := tt.mutations(t, _dependencies.Db)
+
+			organization, err := svc.UpdateOrganization(tt.ctx, updateOrganizationStruct)
+			tt.assertions(t, updateOrganizationStruct, organization, err)
+		})
+	}
+}
+
+type testCaseCreateOrganization struct {
+	name            string
+	ctx             context.Context
+	getDependencies func(t *testing.T) (*dependencies, func(ignoreErrors ...bool))
+	mutations       func(t *testing.T, db *sqlx.DB) (org *model.CreateOrganization)
+	assertions      func(t *testing.T, organization *model.Organization, err error)
+}
+
+func getTestCasesCreateOrganization() []testCaseCreateOrganization {
+	return []testCaseCreateOrganization{
+		{
+			name:            "success",
+			getDependencies: getConcreteDependencies,
+			ctx:             context.TODO(),
+			assertions: func(t *testing.T, organization *model.Organization, err error) {
+				require.NoError(t, err, "unexpected error")
+				require.NotNil(t, organization, "unexpected nil organization")
+
+				require.Equal(t, 1, 1, "unexpected nil organization")
```

Comment:
use require.NotZero() instead of checking if it is 1.

Longer explanation: The organization will have a non-zero ID after a successful insert, in this case we can tell it's 1 because there are no entries, and we are simply seeding it but.. It's more elegant to check it with my suggestion above.

Reply:
I have changed this to use not zero.